src/cache/event.rs

```diff
@@ -337,18 +337,15 @@ impl CacheUpdate for MessageCreateEvent {
         }
 
         let mut messages = cache.messages.entry(self.message.channel_id).or_default();
-        let mut queue = cache.message_queue.entry(self.message.channel_id).or_default();
 
         let mut removed_msg = None;
-
         if messages.len() == max {
-            if let Some(id) = queue.pop_front() {
-                removed_msg = messages.remove(&id);
-            }
+            removed_msg = messages.pop_front();
         }
 
-        queue.push_back(self.message.id);
-        messages.insert(self.message.id, self.message.clone());
+        if !messages.iter().any(|m| m.id == self.message.id) {
```

Comment:
Yes, both to make the tests pass, and because I'm pretty sure discord can deliver duplicate events (although this might be solved by the sequence number).

Reply:
Alright, it can stay. The PR is otherwise fine.